src/transactions/TransactionFrame.cpp

```diff
@@ -235,6 +234,27 @@ TransactionFrame::checkSignatureNoAccount(SignatureChecker& signatureChecker,
     return signatureChecker.checkSignature(signers, 0);
 }
 
+bool
+TransactionFrame::checkExtraSigners(SignatureChecker& signatureChecker)
+{
+    ZoneScoped;
+    if (extraSignersExist())
+    {
+        auto const& extraSigners = mEnvelope.v1().tx.cond.v2().extraSigners;
+        std::vector<Signer> signers;
+
+        std::transform(extraSigners.begin(), extraSigners.end(),
+                       std::back_inserter(signers),
+                       [](SignerKey const& k) { return Signer(k, 1); });
+
+        // We want to verify that there is a signature for each extraSigner, so
+        // we assign a weight of 1 to each key, and set the neededWeight to the
+        // number of extraSigners
+        return signatureChecker.checkSignature(signers, signers.size());
```

Comment:
We should document this in CAP-21. I think from a user outside perspective looking in preventing dupe signers seems to make sense, except that could become a bit of a nasty footgun, if for example you produce a tx that is invalid, or someone tricks you into a tx that is invalid. This is even trickier to guard against because extra signers is keys, not accounts, where-as regular signers on transactions are accounts which could be keys that overlap with extra signers. In fact at transaction build time there would be no way to guarantee an extra signer didn't overlap with account signers already required to sign.

Reply:
If you're just talking about preventing a signer from appearing twice in extra signers, and not overlapping with other tx signers, then sounds fine.

Reply:
@jonjove had identified two scenarios where an dupe signer may appear:
>a duplicate signer in extraSigners
>a signer in extraSigners duplicates a signer from normal signers.

Is this new error condition only for the former, or also the latter?

Reply:
the check is only for the first case (so basically you have twice the same thing in `extraSigners`). We do not check for the second case.